addons/pos_mercado_pago/static/src/app/payment_mercado_pago.js

```diff
@@ -6,40 +6,40 @@ import { AlertDialog } from "@web/core/confirmation_dialog/confirmation_dialog";
 export class PaymentMercadoPago extends PaymentInterface {
     async create_payment_intent() {
         const order = this.pos.get_order();
-        const line = order.selected_paymentline;
+        const line = order.get_selected_paymentline();
         // Build informations for creating a payment intend on Mercado Pago.
         // Data in "external_reference" are send back with the webhook notification
         const infos = {
             amount: parseInt(line.amount * 100, 10),
             additional_info: {
-                external_reference: `${this.pos.config.current_session_id.id}_${line.payment_method.id}_${order.uid}`,
+                external_reference: `${this.pos.config.current_session_id.id}_${line.payment_method_id.id}_${order.pos_reference}`,
```

Comment:
Can you use order UUID ?

```suggestion
                external_reference: `${this.pos.config.current_session_id.id}_${line.payment_method_id.id}_${order.uuid}`,
```

Reply:
I think I could... I read the PR which introduced this very recently https://github.com/odoo/odoo/pull/177300 and he specifically wanted to add `uid/pos_reference` as an identifier so I stayed consistent with this. But I guess the `uuid`could work as well.

I won't be able to test that part though. I don't have access to mercado in Belgium, I'm using credentials from MX. But anyway Yaro says that they have problems with Mercado Pago terminal in Belgium and this behavior cannot be tested.

What do you think?

Reply:
Yes, you can't really test it, but in general in PoS, the unique identifier of an order is the UUID.

Reply:
I'll change that right away then :+1: